src/Illuminate/Support/Pluralizer.php

```diff
@@ -2,99 +2,74 @@
 
 namespace Illuminate\Support;
 
-use Doctrine\Common\Inflector\Inflector;
+use UnexpectedValueException;
+use Illuminate\Support\Pluralizers\EnglishPluralizer;
+use Illuminate\Support\Pluralizers\PluralizerInterface;
 
 class Pluralizer
 {
     /**
-     * Uncountable word forms.
+     * The pluralizer locale.
+     *
+     * @var string
+     */
+    protected static $locale = 'en';
+
+    /**
+     * The registered pluralizers.
      *
      * @var array
      */
-    public static $uncountable = [
-        'audio',
-        'bison',
-        'cattle',
-        'chassis',
-        'compensation',
-        'coreopsis',
-        'data',
-        'deer',
-        'education',
-        'emoji',
-        'equipment',
-        'evidence',
-        'feedback',
-        'firmware',
-        'fish',
-        'furniture',
-        'gold',
-        'hardware',
-        'information',
-        'jedi',
-        'kin',
-        'knowledge',
-        'love',
-        'metadata',
-        'money',
-        'moose',
-        'news',
-        'nutrition',
-        'offspring',
-        'plankton',
-        'pokemon',
-        'police',
-        'rain',
-        'rice',
-        'series',
-        'sheep',
-        'software',
-        'species',
-        'swine',
-        'traffic',
-        'wheat',
-    ];
+    protected static $pluralizers = [];
 
     /**
-     * Get the plural form of an English word.
+     * Register a localized pluralizer.
      *
-     * @param  string  $value
-     * @param  int     $count
-     * @return string
+     * @param string $locale
+     * @param string $pluralizer
+     *
+     * @throws \UnexpectedValueException
      */
-    public static function plural($value, $count = 2)
+    public static function register($locale, $pluralizer)
     {
-        if ((int) $count === 1 || static::uncountable($value)) {
-            return $value;
-        }
+        $pluralizer = is_object($pluralizer) ? get_class($pluralizer) : $pluralizer;
 
-        $plural = Inflector::pluralize($value);
+        if (! in_array(PluralizerInterface::class, class_implements($pluralizer))) {
```

Comment:
Typehind the interface instead of this check.

Reply:
Impossible. Since the Pluralizer's methods are all static it shouldn't be constructed. Hence only a class name is passed and not an instantiated object.

Reply:
How would i go about register a new pluralizer if i'm not supposed to call `Pluralizer::register('da', some::class);` within a service provider? 

If that's how you think of it then we should get rid of this check.

Reply:
This looks the same as `is_a($pluralizer, PluralizerInterface::class, true)` https://secure.php.net/manual/en/function.is-a.php

Reply:
@morloderex You should register a new pluralizer as mentioned, but typehints only accept objects, you're passing a string as well.
@sisve Indeed, will change it. Thanks for catching.